### PR TITLE
feat(cli): add inspect subcommand for torrent metadata

### LINKED ISSUE #20: Add inspect command to display torrent file details
## Feature Description

**What problem does this solve?**
Users need to quickly verify torrent metadata — hash values, piece info, tracker list, file tree — without external tools. Currently there's no way to inspect a `.torrent` file from torrent-builder.

**Proposed Solution**
Add an `inspect` subcommand that reads a `.torrent` file and prints detailed information:

```bash
torrent-builder inspect torrent.torrent
```

**Output should include:**
- Torrent name
- Info hash (v1 and v2 if hybrid)
- Total size
- Piece size and piece count
- File count and file tree
- Tracker list (announce + announce-list)
- Web seeds
- Private flag
- Source field
- Comment
- Creation date
- Created by
- Magnet link

**Optional flags:**
- `--json` for machine-readable output
- `--files` to show detailed file tree only
- `--verify` to check files exist on disk

**Reference:** `mkbrr inspect` provides similar output.

## Who Benefits

All users — essential for verifying torrents before upload, debugging tracker issues, and cross-seeding.

## Acceptance Criteria

- [ ] `inspect` subcommand reads `.torrent` files via `lt::torrent_info`
- [ ] Display all metadata fields listed above
- [ ] Support v1, v2, and hybrid torrents
- [ ] Generate magnet link
- [ ] Unit tests for parsing and formatting
- [ ] Handle corrupted/invalid torrent files gracefully

## Additional Context

- Relatively simple — libtorrent's `lt::torrent_info` exposes all needed data
- Good first feature after the utils module extraction

### PR DESCRIPTION
**Pull request by MadHo3**
Resolving https://github.com/cantalupo555/torrent-builder/issues/20

### CODE REVIEW

**Review by cantalupo555 — COMMENTED** [review]
## Bug: Null pointer dereference in `format_timestamp`

`std::gmtime` can return `nullptr` for out-of-range `time_t` values (e.g., negative, zero, or extremely large timestamps). The result is dereferenced on the next line via `std::put_time(tm_info, ...)` without a null check — this is undefined behavior and will crash on platforms where `gmtime` returns null.

The trigger is realistic: `libtorrent::torrent_info::creation_date()` returns a `time_t` decoded from the torrent's bencode, so a crafted torrent can contain any arbitrary value.

**Fix:** Add a null check:
```cpp
if (tm_info == nullptr) return "Invalid timestamp";
```

**Review by cantalupo555 — COMMENTED** [review]
## Bug: Invalid JSON output for files with symlinks

In `format_file_tree`, the `size_formatted` field has no trailing comma, and the `symlink` field is appended right after. The resulting JSON is:

```json
{
  "path": "...",
  "size": 123,
  "size_formatted": "123 B"
  "symlink": "..."
}
```

Missing comma between `size_formatted` and `symlink` — any JSON parser will reject this.

**Fix:** Add a comma after the `size_formatted` line when `symlink_path` is present.

**Review by cantalupo555 — COMMENTED** [review]
## Missing tests for new functionality

This PR adds 393 lines of new inspector code (`torrent_inspector.cpp`), 6+ new utility functions (`url_encode`, `escape_json`, `format_file_size`, `format_timestamp`, `to_lower`, `starts_with`, `join`, `split`), and a new `inspect` subcommand — but introduces **zero new tests**.

The only test changes are whitespace reformatting of the existing `test_logger.cpp`.

Suggested test coverage:
- Unit tests for each new utility function (edge cases: empty strings, non-ASCII, null timestamps)
- Unit tests for `TorrentInspector::inspect()` with a known torrent file
- Unit tests for `format_file_tree` and `format_metadata` JSON output validity
- Integration test for the `inspect` subcommand CLI (similar to existing CLI integration tests)

**Review by cantalupo555 — COMMENTED** [review]
## `escape_json` does not handle non-ASCII / binary bytes

The function only escapes ASCII control characters (`< 0x20`) and passes all bytes `>= 0x20` through unmodified. Raw non-ASCII bytes (e.g., from filenames with arbitrary binary data in torrent files) will be emitted directly into the JSON string.

While this works for valid UTF-8 filenames, it produces technically invalid JSON per RFC 8259 for arbitrary binary data. Torrent names, comments, and file paths can contain non-UTF-8 bytes.

**Fix options:**
- Escape all bytes >= 0x80 as `\uXXXX` sequences (simple but breaks multi-byte UTF-8)
- Validate UTF-8 sequences and escape invalid ones as `\uXXXX`

**Review by cantalupo555 — COMMENTED** [review]
## Bug: `compute_info_hash` returns v1 hash for hybrid torrents (v1+v2)

`compute_info_hash` uses `if (has_v1()) ... else if (has_v2())` but hybrid torrents have **both**. The `else if` branch is never reached.

At line 131:
```cpp
meta.info_hash_v2 = info_hash.has_v2() ? compute_info_hash(info_hash) : "";
```

For hybrids: `has_v2()` is `true`, but `compute_info_hash` enters the `has_v1()` branch → returns the v1 SHA-1 hash (40 chars) instead of the v2 SHA-256 hash (64 chars).

The magnet link is also affected — `xt=urn:btmh:` gets the wrong v1 hash.

**Fix:** Split into two dedicated methods that access `hash.v1` and `hash.v2` directly:
```cpp
std::string compute_info_hash_v1(const libtorrent::info_hash_t &hash) const {
    // iterate hash.v1
}
std::string compute_info_hash_v2(const libtorrent::info_hash_t &hash) const {
    // iterate hash.v2
}
```

File: src/torrent_inspector.cpp [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+        std::stringstream json;
+        json << "{\n  \"files\": [\n";
+        for (size_t i = 0; i < meta.files.size(); ++i)
+        {
+            const auto &file = meta.files[i];
+            json << "    {\n";
+            json << "      \"path\": \"" << utils::escape_json(file.path) << "\",\n";
+            json << "      \"size\": " << file.size << ",\n";
+            json << "      \"size_formatted\": \"" << utils::format_file_size(file.size) << "\"\n";
+            if (file.symlink_path)
```
Comment (cantalupo555) [inline src/torrent_inspector.cpp:368]: Missing comma after `size_formatted` when `symlink_path` is present. Produces invalid JSON.

File: src/torrent_inspector.cpp [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+    ss << std::hex << std::setfill('0');
+
+    if (hash.has_v1())
+    {
+        for (unsigned char byte : hash.v1)
+        {
+            ss << std::setw(2) << static_cast<int>(byte);
+        }
+    }
+    else if (hash.has_v2())
```
Comment (cantalupo555) [inline src/torrent_inspector.cpp:65]: `else if (hash.has_v2())` is never reached for hybrid (v1+v2) torrents. The `if (hash.has_v1())` above captures the case first.

File: src/torrent_builder.cpp [hunk 3]
```diff
@@ -411,87 +514,177 @@ std::optional<TorrentConfig> get_commandline_config(const cxxopts::ParseResult&
...
-        );
+        throw std::runtime_error("Error accessing path '" + e.path1().string() + "': " + e.what());
     }
 }
 
+int handle_inspect_command(const std::vector<std::string> &args)
+{
+    try
+    {
+        // ساخت آرایه از رشته‌ها
```
Comment (cantalupo555) [inline src/torrent_builder.cpp:546]: This Farsi comment is inconsistent with the rest of the codebase which uses English comments exclusively. Should be translated or removed.

File: src/utils.cpp [hunk 4]
```diff
@@ -48,40 +72,196 @@ std::string sanitize_path(std::string input) {
...
+    return formatted.str();
+}
+
+std::string format_timestamp(int64_t timestamp)
+{
+    std::time_t time = static_cast<std::time_t>(timestamp);
+    std::tm *tm_info = std::gmtime(&time);
+
+    std::ostringstream formatted;
+    formatted << std::put_time(tm_info, "%Y-%m-%d %H:%M:%S") << " UTC";
```
Comment (cantalupo555) [inline src/utils.cpp:214]: `std::gmtime` can return `nullptr` for out-of-range values. Dereferencing without a null check is UB / crash.

**Review by cantalupo555 — COMMENTED** [review]
## Bug: `compute_info_hash` returns v1 hash for hybrid torrents (v1+v2)

`compute_info_hash` uses `if (has_v1()) ... else if (has_v2())` but hybrid torrents have **both**. The `else if` branch is never reached.

At line 131:
```cpp
meta.info_hash_v2 = info_hash.has_v2() ? compute_info_hash(info_hash) : "";
```

For hybrids: `has_v2()` is `true`, but `compute_info_hash` enters the `has_v1()` branch → returns the v1 SHA-1 hash (40 chars) instead of the v2 SHA-256 hash (64 chars).

The magnet link is also affected — `xt=urn:btmh:` gets the wrong v1 hash.

**Fix:** Split into two dedicated methods that access `hash.v1` and `hash.v2` directly:
```cpp
std::string compute_info_hash_v1(const libtorrent::info_hash_t &hash) const {
    // iterate hash.v1
}
std::string compute_info_hash_v2(const libtorrent::info_hash_t &hash) const {
    // iterate hash.v2
}
```

**Review by cantalupo555 — COMMENTED** [review]
## Bug: Null pointer dereference in `format_timestamp`

`std::gmtime` can return `nullptr` for out-of-range `time_t` values (e.g., negative, zero, or extremely large timestamps). The result is dereferenced on the next line via `std::put_time(tm_info, ...)` without a null check — this is undefined behavior and will crash on platforms where `gmtime` returns null.

The trigger is realistic: `libtorrent::torrent_info::creation_date()` returns a `time_t` decoded from the torrent's bencode, so a crafted torrent can contain any arbitrary value.

**Fix:** Add a null check:
```cpp
if (tm_info == nullptr) return "Invalid timestamp";
```

**Review by cantalupo555 — COMMENTED** [review]
## Bug: Invalid JSON output for files with symlinks

In `format_file_tree`, the `size_formatted` field has no trailing comma, and the `symlink` field is appended right after. The resulting JSON is:

```json
{
  "path": "...",
  "size": 123,
  "size_formatted": "123 B"
  "symlink": "..."
}
```

Missing comma between `size_formatted` and `symlink` — any JSON parser will reject this.

**Fix:** Add a comma after the `size_formatted` line when `symlink_path` is present.

File: src/utils.cpp [hunk 5]
```diff
@@ -48,40 +72,196 @@ std::string sanitize_path(std::string input) {
...
+            break;
+        default:
+            if (static_cast<unsigned char>(c) < 0x20)
+            {
+                escaped << "\\u" << std::hex << std::setw(4) << std::setfill('0')
+                        << static_cast<int>(c);
+            }
+            else
+            {
+                escaped << c;
```
Comment (cantalupo555) [inline src/utils.cpp:182]: Bytes >= 0x80 (non-ASCII) pass through unescaped. Produces invalid JSON per RFC 8259 for binary data.

**Review by cantalupo555 — COMMENTED** [review]
## `escape_json` does not handle non-ASCII / binary bytes

The function only escapes ASCII control characters (`< 0x20`) and passes all bytes `>= 0x20` through unmodified. Raw non-ASCII bytes (e.g., from filenames with arbitrary binary data in torrent files) will be emitted directly into the JSON string.

While this works for valid UTF-8 filenames, it produces technically invalid JSON per RFC 8259 for arbitrary binary data. Torrent names, comments, and file paths can contain non-UTF-8 bytes.

**Fix options:**
- Escape all bytes >= 0x80 as `\uXXXX` sequences (simple but breaks multi-byte UTF-8)
- Validate UTF-8 sequences and escape invalid ones as `\uXXXX`
